lib/vmpooler/api/v1.rb

```diff
@@ -120,6 +120,74 @@ def atomically_allocate_vms(payload)
       result
     end
 
+    def update_pool_size(payload)
+      result = { 'ok' => false }
+
+      pool_index = pool_index(pools)
+      pools_updated = 0
+      sync_pool_sizes
+
+      payload.each do |poolname, size|
+        unless pools[pool_index[poolname]]['size'] == size.to_i
+          pools[pool_index[poolname]]['size'] = size.to_i
+          backend.hset('vmpooler__config__poolsize', poolname, size)
```

Comment:
It's because redis always returns a string value for the 'vmpooler__config__poolsize' entry, whether your input was an integer, or not.

Reply:
Ah that sucks. You could still DRY this up by something like
```
update_pool_config_entry(payload, entry, munge = lambda { |x| x })
```

so that we would have:
```
update_pool_size(payload) = update_pool_config_entry(payload, 'size', lambda { |size| size.to_i })
```
```
update_pool_template(payload) = update_pool_config_entry(payload, 'template')
```

and you can do something similar for the sync. methods.

(The similarity between the update methods is also what lead my Q. about the single `config/pool` endpoint).